src/CoreWCF.Primitives/src/CoreWCF/Channels/ChannelBuilder.cs

```diff
@@ -47,25 +47,29 @@ public ChannelDemuxer ChannelDemuxer
         {
             get { return channelDemuxer; }
         }
+        public TypedChannelDemuxer GetTypedChannelDemuxer<TChannel>() where TChannel : class, IChannel
+        {
+            return ChannelDemuxer.GetTypedServiceDispatcher<TChannel>(context);
+        }
         public  IServiceDispatcher AddServiceDispatcher<TChannel>(IServiceDispatcher innerDispatcher) where TChannel : class, IChannel
         {
             if (!isChannelDemuxerRequired)
                 throw new Exception("ChannelDemuxerRequired is set to false");
-            return channelDemuxer.CreaterServiceDispatcher<TChannel>(innerDispatcher);
+            return channelDemuxer.CreaterServiceDispatcher<TChannel>(innerDispatcher, context);
```

Comment:
>CreaterServiceDispatcher [](start = 34, length = 24)

I just noticed this method is called Create*r*ServiceDispatcher. Can you fix the name

Reply:
done